### ISSUE TITLE
Inconsistent values when dividing by negative zero

### ISSUE THREAD

**Issue by tinkerer-red** [YoYoGames/GameMaker-Bugs#8115]
### Description

There is a strange result which occurs from dividing by zero, given that there is no such thing as `-0` in real world mathematics, it would make sense that all of these would return `infinity`. But also this wouldn't be IEEE-754 standard, but if we kept with the standard then we have no way of printing or stringing a negative zero.
![Image](https://github.com/user-attachments/assets/1a7fded2-88d5-4ca2-be46-44ee5d343057)

So the issue will probably need to be updated with what ever YoYo team thinks they would like to abide by.
Either:
- dividing by `-0` produces `-infinity`
**or**
- compile time optimization fails with `-0`
- `string()` cant print negative zero

### Expected Change

To be decided by YoYo

### Steps To Reproduce

```gml
show_debug_message("\n\n\n")

var _zero = 0;
var _neg_zero = -0;
var _post_compile_neg_zero = -_zero;
show_debug_message(1/0) // Output :: inf
show_debug_message(1/-0) // Output :: -inf
show_debug_message(1/_zero) // Output :: inf
show_debug_message(1/ -_zero) // Output :: -inf
show_debug_message(1/_neg_zero) // Output :: inf
show_debug_message(1/_post_compile_neg_zero) // Output :: -inf

show_debug_message(1*0) // Output :: 0
show_debug_message(1*-0) // Output :: 0
show_debug_message(1*_zero) // Output :: 0
show_debug_message(1* -_zero) // Output :: 0
show_debug_message(1*_neg_zero) // Output :: 0
show_debug_message(1*_post_compile_neg_zero) // Output :: 0

show_debug_message(0) // Output :: 0
show_debug_message(-0) // Output :: 0
show_debug_message(-_zero) // Output :: 0
show_debug_message(-_zero) // Output :: 0
show_debug_message(_neg_zero) // Output :: 0
show_debug_message(_post_compile_neg_zero) // Output :: 0

show_debug_message("\n\n\n")
```

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.8.1 (Monthly)

### Which platform(s) are you seeing the problem on?

Windows

### Sample Package Attached?

- [x] I have attached my bug package

### Sample Project Added?

- [x] I have included a small sample project

**Comment by rwkay** [comment]
There is no such thing as -0 though, 0 cannot be signed... there is such a thing as -infinity though... 

Sorry this looks consistent to me.

**Comment by tinkerer-red** [comment]
`_neg_zero` produces different results then `-0` as a literal does though.

It does appear `-0` is valid in gml from this test, unsure if thats a long running bug though.
![Image](https://github.com/user-attachments/assets/be0c0215-5c61-4495-8cf9-8ab38a64e662)
```gml
var _compile_time = 1/-infinity;
var _temp = -infinity;
var _runtime = 1/_temp;

show_debug_message(_compile_time)
show_debug_message(_runtime)
show_debug_message(1/_compile_time)
show_debug_message(1/_runtime)
```

**Comment by rwkay** [comment]
I do not think that there is a bug here... it is working as the IEEE spec says - see https://en.wikipedia.org/wiki/Division_by_zero#:~:text=Dividing%20any%20non-zero%20number,in%20case%20of%20arithmetic%20underflow.

Closing as not a bug.

It is not how you are expecting it... but it is correct.

**Comment by tinkerer-red** [comment]
That's fair enough, it's not much of an issue anyways
